spec/codeowners/cli/owners_list_handler_spec.rb

```diff
@@ -0,0 +1,197 @@
+# frozen_string_literal: true
+
+RSpec.describe Codeowners::Cli::OwnersListHandler do
+  let(:owners_list_handler) { described_class.new }
+
+  describe '#suggest_add_to_owners_list' do
+    let(:new_file) { 'test.rb' }
+    let(:the_owner) { '@owner1' }
+    let(:another_owner1) { '@owner2' }
+    let(:another_owner2) { '@owner3' }
+    let(:line_str) { "#{new_file} #{the_owner}" }
+    let(:line) { Codeowners::Checker::Group::Line.build(line_str) }
+
+    before do
+      allow(owners_list_handler).to receive(:add_to_ownerslist_dialog)
+    end
+
+    context 'when ignored list is empty' do
+      it 'calls #add_to_ownerslist_dialog' do
+        owners_list_handler.suggest_add_to_owners_list(line, the_owner)
+
+        expect(owners_list_handler).to have_received(:add_to_ownerslist_dialog).with(line, the_owner)
+      end
+    end
+
+    context 'when ignored list contains provided owner' do
+      let(:ingored_owners) { [another_owner1, the_owner] }
+
+      before do
+        allow(owners_list_handler).to receive(:ignored_owners).and_return(ingored_owners)
+      end
+
+      it 'does not call #add_to_ownerslist_dialog and return nil' do
+        output = owners_list_handler.suggest_add_to_owners_list(line, the_owner)
+
+        expect(output).to be_nil
+        expect(owners_list_handler).not_to have_received(:add_to_ownerslist_dialog).with(line, the_owner)
+      end
+    end
+
+    context 'when ignored list does not contain provided owner' do
+      let(:ingored_owners) { [another_owner1, another_owner2] }
+
+      before do
+        allow(owners_list_handler).to receive(:ignored_owners).and_return(ingored_owners)
```

Comment:
Seems like this `before` setup is duplicated with the parent one on line 15. Removing this block make the code passes in the same way.

Reply:
Parent's `before` on line 15 mocks `add_to_ownerslist_dialog`. On the other hand line 30 looks similar indeed but that is a sibling not a parent and other siblings do not need that mocking so it doesn't make sense to put it in the parent I think.

Removing this block works just because empty list also does not contain `the_owner` and behaves in the same way (which is more explicitly stated by another variant of specs below that use `shared_examples` to emphasize the fact).